idp_task1.cpp

```diff
@@ -58,8 +61,66 @@ void createTestTCPServer()
     }
 }
 
+void createTestAsyncTCPServer()
+{
+    try
+    {
+        fd_set read_fdset;
+
+        timeval timeout;
+        timeout.tv_sec = 1;
+        timeout.tv_usec = 0;
+
+        Socket_sptr master_server_socket = std::make_shared<Socket>(AF_INET, SOCK_STREAM, 0);
+
+        int maxfd = master_server_socket->GetSocketHandle();
+
+        master_server_socket->Bind(INADDR_ANY, SERVER_PORT);
+
+        master_server_socket->Listen(LISTENQ);
+
+        AsyncTCPServer server(master_server_socket, read_fdset);
+
+        while(true)
+        {
+            Socket_vector ready_sockets = server.Select(maxfd + 1, &timeout);
+
+            for(auto socket : ready_sockets)
+            {
+                if(socket->GetSocketHandle() == master_server_socket->GetSocketHandle())
+                {
+                    Socket_sptr accepted_socket = server.AcceptIncomingConnection();
+
+                    if(maxfd < accepted_socket->GetSocketHandle())
+                        maxfd = accepted_socket->GetSocketHandle();
```

Comment:
Ooow, you're using it that way. :)
Okay, I was wrong, your select should work, but very inoptimal.